code/ATMOSPHERICS/atmospherics.dm

```diff
@@ -273,36 +276,61 @@ GLOBAL_DATUM_INIT(pipe_icon_manager, /datum/pipe_icon_manager, new())
 
 	var/obj/machinery/atmospherics/target_move = findConnecting(direction)
 	if(target_move)
-		if(is_type_in_list(target_move, GLOB.ventcrawl_machinery) && target_move.can_crawl_through())
-			user.remove_ventcrawl()
-			user.forceMove(target_move.loc) //handles entering and so on
-			user.visible_message("You hear something squeezing through the ducts.", "You climb out the ventilation system.")
+		if(is_type_in_list(target_move, GLOB.ventcrawl_machinery))
+			if(target_move.can_crawl_through())
+				if(do_after(user, 30, target = target_move))
+					user.remove_ventcrawl()
+					user.forceMove(target_move.loc) //handles entering and so on
+					user.visible_message("You hear something squeezing through the ducts.", "You climb out the ventilation system.")
+			else if(istype(user, /mob/living/silicon/robot/drone))
+				var/mob/living/silicon/robot/drone/D = user
+				if(D.emagged)
+					return
```

Comment:
The check to make emagged drones *unable* to break out of welded vents feels wrong. If anything, it should be the other way around.

Reply:
This prevention has been removed.